salt/log/handlers/__init__.py

```diff
@@ -162,135 +169,28 @@ class WatchedFileHandler(ExcInfoOnLogLevelFormatMixIn, logging.handlers.WatchedF
-    #  https://bugs.python.org/issue35726
-    class QueueHandler(ExcInfoOnLogLevelFormatMixIn, logging.handlers.QueueHandler):  # pylint: disable=no-member,E0240
-
-        def enqueue(self, record):
-            '''
-            Enqueue a record.
-
-            The base implementation uses put_nowait. You may want to override
-            this method if you want to use blocking, timeouts or custom queue
-            implementations.
-            '''
-            try:
-                self.queue.put_nowait(record)
-            except queue.Full:
-                sys.stderr.write('[WARNING ] Message queue is full, '
-                                 'unable to write "{0}" to log'.format(record))
-
-        def prepare(self, record):
-            '''
-            Prepares a record for queuing. The object returned by this method is
-            enqueued.
-            The base implementation formats the record to merge the message
-            and arguments, and removes unpickleable items from the record
-            in-place.
-            You might want to override this method if you want to convert
-            the record to a dict or JSON string, or send a modified copy
-            of the record while leaving the original intact.
-            '''
-            # The format operation gets traceback text into record.exc_text
-            # (if there's exception data), and also returns the formatted
-            # message. We can then use this to replace the original
-            # msg + args, as these might be unpickleable. We also zap the
-            # exc_info and exc_text attributes, as they are no longer
-            # needed and, if not None, will typically not be pickleable.
-            msg = self.format(record)
-            # bpo-35726: make copy of record to avoid affecting other handlers in the chain.
-            record = copy.copy(record)
-            record.message = msg
-            record.msg = msg
-            record.args = None
-            record.exc_info = None
-            record.exc_text = None
-            return record
-else:
-    class QueueHandler(ExcInfoOnLogLevelFormatMixIn, logging.handlers.QueueHandler):  # pylint: disable=no-member,E0240
-
-        def enqueue(self, record):
-            '''
-            Enqueue a record.
-
-            The base implementation uses put_nowait. You may want to override
-            this method if you want to use blocking, timeouts or custom queue
-            implementations.
-            '''
-            try:
-                self.queue.put_nowait(record)
-            except queue.Full:
-                sys.stderr.write('[WARNING ] Message queue is full, '
-                                 'unable to write "{0}" to log'.format(record))
+        Writes the LogRecord to the queue, preparing it for pickling first.
+        '''
+        try:
+            self.sender.send(self.prepare(record))
+        except Exception:
+            self.handleError(record)
```

Comment:
You seem to have removed the queue handlers, 3rd-party code interacting with salt or using salt as a lib could depend on it.
Let's add it back?

Reply:
I added it back.